src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -158,6 +158,17 @@ public void LoggerProviderException(string methodName, Exception ex)
             }
         }
 
+        [NonEvent]
+        public void LoggerProcessStateSkipped<TState>()
+        {
+            if (this.IsEnabled(EventLevel.Warning, EventKeywords.All))
+            {
+                this.LoggerProcessStateSkipped(
+                    typeof(TState).FullName!,
+                    "because it does not implement either IReadOnlyList<KeyValuePair<string, object>> or IEnumerable<KeyValuePair<string, object>>");
```

Comment:
nit:
```suggestion
                    "because it implements neither IReadOnlyList<KeyValuePair<string, object>> nor IEnumerable<KeyValuePair<string, object>>");
```

Reply:
I tweaked it. Let me know what you think of the new version. I think your suggestion is correct I'm just trying to keep it a bit more colloquial.